gravatar/src/main/java/com/gravatar/services/AvatarService.kt

```diff
@@ -22,51 +16,35 @@ public class AvatarService(private val okHttpClient: OkHttpClient? = null) {
         const val LOG_TAG = "AvatarService"
     }
 
-    // Run onResponse and onError callbacks on the main thread
-    private val coroutineScope = CoroutineScope(GravatarSdkDI.dispatcherMain)
-
     /**
      * Uploads a Gravatar image for the given email address.
      *
      * @param file The image file to upload
      * @param email The email address to associate the image with
      * @param wordpressBearerToken The bearer token for the user's WordPress/Gravatar account
-     * @param gravatarUploadListener The listener to notify of the upload result
      */
-    public fun upload(
-        file: File,
-        email: Email,
-        wordpressBearerToken: String,
-        gravatarUploadListener: GravatarListener<Unit, ErrorType>,
-    ) {
+    public suspend fun upload(file: File, email: Email, wordpressBearerToken: String): Result<Unit, ErrorType> {
```

Comment:
I would define this main safe as recommended in the [coroutines best practices doc](https://developer.android.com/kotlin/coroutines/coroutines-best-practices#main-safe)